qiskit_experiments/randomized_benchmarking/rb_analysis.py

```diff
@@ -0,0 +1,155 @@
+    level2_probability,
+    mean_xy_data,
+    filter_data,
+)
+from qiskit_experiments.analysis.plotting import plot_curve_fit
+
+try:
+    from matplotlib import pyplot as plt
+
+    HAS_MATPLOTLIB = True
+except ImportError:
+    HAS_MATPLOTLIB = False
+
+
+class RBAnalysis(BaseAnalysis):
+    """RB Analysis class."""
+
+    # pylint: disable = arguments-differ, invalid-name, attribute-defined-outside-init
+    def _run_analysis(
+        self,
+        experiment_data,
+        p0: Optional[List[float]] = None,
+        plot: bool = True,
+        ax: Optional["AxesSubplot"] = None,
+    ):
+        """Run analysis on circuit data.
+        Args:
+            experiment_data (ExperimentData): the experiment data to analyze.
+            p0: Optional, initial parameter values for curve_fit.
+            plot: If True generate a plot of fitted data.
+            ax: Optional, matplotlib axis to add plot to.
+        Returns:
+            tuple: A pair ``(analysis_result, figures)`` where
+                   ``analysis_results`` may be a single or list of
+                   AnalysisResult objects, and ``figures`` may be
+                   None, a single figure, or a list of figures.
+        """
+        self._num_qubits = len(experiment_data.data[0]["metadata"]["qubits"])
+        xdata, ydata, ydata_sigma = self._extract_data(experiment_data)
+
+        def fit_fun(x, a, alpha, b):
+            return a * alpha ** x + b
+
+        p0 = self._p0(xdata, ydata)
+        analysis_result = curve_fit(
+            fit_fun,
+            xdata,
+            ydata,
+            p0,
+            ydata_sigma,
+            bounds=([0, 0, 0], [1, 1, 1]),
+            absolute_sigma=False,
+        )
+
+        # Add EPC data
+        popt = analysis_result["popt"]
+        popt_err = analysis_result["popt_err"]
+        scale = (2 ** self._num_qubits - 1) / (2 ** self._num_qubits)
+        analysis_result["EPC"] = scale * (1 - popt[1])
+        analysis_result["EPC_err"] = scale * popt_err[1] / popt[1]
+        analysis_result["plabels"] = ["A", "alpha", "B"]
+
+        if plot:
+            fig = plot_curve_fit(fit_fun, analysis_result, ax=ax)
+            self._format_plot(fig, analysis_result)
+            analysis_result.plt = plt
+        return analysis_result, None
+
+    def _p0(self, xdata, ydata):
+        fit_guess = [0.95, 0.99, 1 / 2 ** self._num_qubits]
+        # Use the first two points to guess the decay param
+        dcliff = xdata[1] - xdata[0]
+        dy = (ydata[1] - fit_guess[2]) / (ydata[0] - fit_guess[2])
+        alpha_guess = dy ** (1 / dcliff)
+        if alpha_guess < 1.0:
+            fit_guess[1] = alpha_guess
+
+        if ydata[0] > fit_guess[2]:
+            fit_guess[0] = (ydata[0] - fit_guess[2]) / fit_guess[1] ** xdata[0]
+
+        return fit_guess
+
+    def _extract_data(self, experiment_data, **filters):
+        """Extract the base data for the fitter from the experiment data.
```

Comment:
To my understanding at least some of this will be done by #19 and #22

Reply:
Most of the work is already done in #19; when #22 is merged we can check how to proceed.